{"name": "test_invalid_login", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x560ca4573d0a <unknown>\n#1 0x560ca40245f0 <unknown>\n#2 0x560ca4075a33 <unknown>\n#3 0x560ca4075c21 <unknown>\n#4 0x560ca40c4274 <unknown>\n#5 0x560ca409b68d <unknown>\n#6 0x560ca40c1660 <unknown>\n#7 0x560ca409b433 <unknown>\n#8 0x560ca4067ea3 <unknown>\n#9 0x560ca4068b01 <unknown>\n#10 0x560ca4538b5b <unknown>\n#11 0x560ca453ca41 <unknown>\n#12 0x560ca451fc52 <unknown>\n#13 0x560ca453d5b4 <unknown>\n#14 0x560ca4503f0f <unknown>\n#15 0x560ca4561db8 <unknown>\n#16 0x560ca4561f96 <unknown>\n#17 0x560ca4572b56 <unknown>\n#18 0x7fe75189caa4 <unknown>\n#19 0x7fe751929c3c <unknown>", "trace": "self = <tests.test_login.TestLogin object at 0x7f4e51ed5e00>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"faa7dfdeac224defbc4af542957d9221\")>\n\n    @pytest.mark.selenium\n    def test_invalid_login(self, driver):\n        driver.get(\"https://www.saucedemo.com/\")\n        login_page = LoginPage(driver)\n        login_page.login(\"standarduser\", \"secret_sauce\")\n>       assert login_page.is_error_displayed(), \"Error Should be visible\"\n\ntests/test_login.py:26: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/login_page.py:30: in is_error_displayed\n    return self.wait_for_visibility(self.login_error_text_alert)\npages/base_page.py:26: in wait_for_visibility\n    return self.wait.until(expected_conditions.visibility_of_element_located(by_locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"faa7dfdeac224defbc4af542957d9221\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x7f4e51e337e0>\nmessage = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Wait until the method returns a value that is not False.\n    \n        Calls the method provided with the driver as an argument until the\n        return value does not evaluate to ``False``.\n    \n        Parameters:\n        ----------\n        method: callable(WebDriver)\n            - A callable object that takes a WebDriver instance as an argument.\n    \n        message: str\n            - Optional message for :exc:`TimeoutException`\n    \n        Return:\n        -------\n        object: T\n            - The result of the last call to `method`\n    \n        Raises:\n        -------\n        TimeoutException\n            - If 'method' does not return a truthy value within the WebDriverWait\n            object's timeout\n    \n        Example:\n        --------\n        >>> from selenium.webdriver.common.by import By\n        >>> from selenium.webdriver.support.ui import WebDriverWait\n        >>> from selenium.webdriver.support import expected_conditions as EC\n    \n        # Wait until an element is visible on the page\n        >>> wait = WebDriverWait(driver, 10)\n        >>> element = wait.until(EC.visibility_of_element_located((By.ID, \"exampleId\")))\n        >>> print(element.text)\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            if time.monotonic() > end_time:\n                break\n            time.sleep(self._poll)\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x560ca4573d0a <unknown>\nE       #1 0x560ca40245f0 <unknown>\nE       #2 0x560ca4075a33 <unknown>\nE       #3 0x560ca4075c21 <unknown>\nE       #4 0x560ca40c4274 <unknown>\nE       #5 0x560ca409b68d <unknown>\nE       #6 0x560ca40c1660 <unknown>\nE       #7 0x560ca409b433 <unknown>\nE       #8 0x560ca4067ea3 <unknown>\nE       #9 0x560ca4068b01 <unknown>\nE       #10 0x560ca4538b5b <unknown>\nE       #11 0x560ca453ca41 <unknown>\nE       #12 0x560ca451fc52 <unknown>\nE       #13 0x560ca453d5b4 <unknown>\nE       #14 0x560ca4503f0f <unknown>\nE       #15 0x560ca4561db8 <unknown>\nE       #16 0x560ca4561f96 <unknown>\nE       #17 0x560ca4572b56 <unknown>\nE       #18 0x7fe75189caa4 <unknown>\nE       #19 0x7fe751929c3c <unknown>\n\n/opt/hostedtoolcache/Python/3.13.3/x64/lib/python3.13/site-packages/selenium/webdriver/support/wait.py:146: TimeoutException"}, "steps": [{"name": "login", "status": "passed", "steps": [{"name": "fill_username", "status": "passed", "parameters": [{"name": "username", "value": "'standarduser'"}], "start": 1744908272986, "stop": 1744908273047}, {"name": "fill_password", "status": "passed", "parameters": [{"name": "password", "value": "'secret_sauce'"}], "start": 1744908273047, "stop": 1744908273112}, {"name": "click_login", "status": "passed", "start": 1744908273112, "stop": 1744908273162}], "parameters": [{"name": "username", "value": "'standarduser'"}, {"name": "password", "value": "'secret_sauce'"}], "start": 1744908272986, "stop": 1744908273162}, {"name": "is_error_displayed", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x560ca4573d0a <unknown>\n#1 0x560ca40245f0 <unknown>\n#2 0x560ca4075a33 <unknown>\n#3 0x560ca4075c21 <unknown>\n#4 0x560ca40c4274 <unknown>\n#5 0x560ca409b68d <unknown>\n#6 0x560ca40c1660 <unknown>\n#7 0x560ca409b433 <unknown>\n#8 0x560ca4067ea3 <unknown>\n#9 0x560ca4068b01 <unknown>\n#10 0x560ca4538b5b <unknown>\n#11 0x560ca453ca41 <unknown>\n#12 0x560ca451fc52 <unknown>\n#13 0x560ca453d5b4 <unknown>\n#14 0x560ca4503f0f <unknown>\n#15 0x560ca4561db8 <unknown>\n#16 0x560ca4561f96 <unknown>\n#17 0x560ca4572b56 <unknown>\n#18 0x7fe75189caa4 <unknown>\n#19 0x7fe751929c3c <unknown>\n\n", "trace": "  File \"/opt/hostedtoolcache/Python/3.13.3/x64/lib/python3.13/site-packages/allure_commons/_allure.py\", line 202, in impl\n    return func(*a, **kw)\n  File \"/home/runner/work/python-automation-framework/python-automation-framework/pages/login_page.py\", line 30, in is_error_displayed\n    return self.wait_for_visibility(self.login_error_text_alert)\n           ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/runner/work/python-automation-framework/python-automation-framework/pages/base_page.py\", line 26, in wait_for_visibility\n    return self.wait.until(expected_conditions.visibility_of_element_located(by_locator))\n           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.3/x64/lib/python3.13/site-packages/selenium/webdriver/support/wait.py\", line 146, in until\n    raise TimeoutException(message, screen, stacktrace)\n"}, "start": 1744908273162, "stop": 1744908293207}], "attachments": [{"name": "Failure Screenshot", "source": "44bb7a2a-09de-459c-b5c2-a2aa6c62167b-attachment.png", "type": "image/png"}], "start": 1744908272933, "stop": 1744908293209, "uuid": "76cfa29b-893f-4496-b7de-ec91c0086e3c", "historyId": "5ad05e7b3e746de3eeebc71b61c34a27", "testCaseId": "5ad05e7b3e746de3eeebc71b61c34a27", "fullName": "tests.test_login.TestLogin#test_invalid_login", "labels": [{"name": "tag", "value": "selenium"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_login"}, {"name": "subSuite", "value": "TestLogin"}, {"name": "host", "value": "fv-az1950-551"}, {"name": "thread", "value": "2881-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_login"}]}